multipatch_analysis/ui/acq4_modules.py

```diff
@@ -56,3 +71,266 @@ def load_from_dm_clicked(self):
+
+        self.create_json_btn = QtGui.QPushButton("Save Cell Location Json")
+        self.layout.addWidget(self.create_json_btn, 1, 0)
+        self.create_json_btn.clicked.connect(self.create_lims_json)
+
+        self.create_test_btn = QtGui.QPushButton('Testing Button')
+        self.layout.addWidget(self.create_test_btn, 2, 0)
+        self.create_test_btn.clicked.connect(self.test_button)
+
+    def test_button(self):
+        """place to test things
+        """
+        items = self.mosaic_editor.canvas.items 
+        markers = [i for i in items if isinstance(i, MarkersCanvasItem)]
+        if len(markers) != 1:
+            raise Exception("Must have exactly 1 Markers item in the canvas.")
+        pipettes = markers[0].saveState()['markers']
+        for cell in pipettes:
+            p = cell[1]
+            for i in items:
+                try:
+                    if i.opts['handle'].name() == self.image_20:
+                        image = i
+                except AttributeError:
+                    pass
+            print(markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).x())
+            print(markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).y())
+
+        
+
+    
+
+    def load_clicked(self):
+        """
+        Checks that directory is pointed at a slice
+        Check LIMS for 20x biocytin image and if 20x image is saved locally.
+        If image not saved locally then copies file from LIMS.
+        Opens button to create a JSON of the cell locations
+        """
+        self.base_dir = self.mosaic_editor.ui.fileLoader.baseDir()
+        if self.base_dir.info()['dirType'] == 'Slice':
+            self.base_path = self.base_dir.name()
+            self.slice_name = self.base_dir.info()['specimen_ID']
+            self.slice_id = lims.specimen_id_from_name(self.slice_name)
+            clusters = lims.cell_cluster_ids(self.slice_id)
+            """if len(clusters) == 0:
+                                                    raise Exception("No Clusters Recorded in LIMS")"""
+            try:
+                aff_image_path = lims.specimen_20x_image(self.slice_name)
+                #check the image file path
+                safe_aff_image_path = lims.lims.safe_system_path(aff_image_path)
+
+                if os.path.exists(safe_aff_image_path) == True:
+                    print('File Exists in LIMS')
+                    aff_image_name = aff_image_path.split("/")[-1]
+                    jpg_image_name = aff_image_name.split(".aff")[0] + ".jpg"
+
+                    image_service_url = "http://lims2/cgi-bin/imageservice?path="
+                    zoom = 5        #need to check that this ACQ4 can support this size
+
+                    full_url = image_service_url + aff_image_path + "&zoom={}".format(str(zoom))
+                    print(full_url)
+                    save_path = self.base_path + '/' + jpg_image_name
```

Comment:
Not safe to manually generate these paths because different systems require '\\' or '/'. Instead use `os.path.join`.

Reply:
I missed that one when switching the other ones over.